src/service/tcp.rs

```diff
@@ -0,0 +1,149 @@
+use async_trait::async_trait;
+use axum::{
+    extract::{rejection::JsonRejection, FromRequestParts, State},
+    http::{header, request::Parts, StatusCode},
+    response::{IntoResponse, Json},
+};
+use axum_macros::debug_handler;
+use hyper::upgrade::OnUpgrade;
+use tracing::{debug, error, info, trace};
+use uuid::Uuid;
+
+use crate::{
+    domain::notary::{ClientType, NotarizationRequest, NotarizationResponse, NotarizationSetup},
+    service::notary_service,
+    NotaryServerError,
+};
+
+/// Custom extractor used to extract underlying TCP connection for TCP client — using the same upgrade primitives used by
```

Comment:
This is the trick that I mentioned in my update in [Discord](https://discord.com/channels/974410303255232542/987068645630947379/1135947819694436362) — to combine both the handlers of /notarize and /ws-notarize using one axum router and one hyper protocol. The only material impact to the TCP client is that they need to add an extra header "Upgrade: TCP" in their request for this to work.

Reply:
Can refer to the TCP client implementation here: https://github.com/tlsnotary/notary-server/blob/NS-2_http_websocket/tests/integration_test.rs#L126-L136